opal/mca/hwloc/base/hwloc_base_util.c

```diff
@@ -1066,7 +1066,7 @@ static char *bitmap2rangestr(int bitmap)
         } else {
             snprintf(tmp, stmp, "%d-%d", range_start, range_end);
         }
-        strncat(ret, tmp, sizeof(ret) - strlen(ret) - 1);
+        snprintf(ret + strlen(ret), BUFSIZ, "%s",  tmp);
```

Comment:
2nd arg should be `BUFSIZ - strlen(ret)`.  While looking at, would it be better to use `OMPI_AFFINITY_STRING_MAX` instead of `BUFSIZ` ?

Reply:
These strings use `BUFSIZ` - so it's possible we probably should use OMPI_AFFINITY_STRING_MAX....perhaps more cleanup to do.